core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3092,6 +3096,36 @@ private static class ArrayConcatImplementor extends AbstractRexCallImplementor {
     }
   }
 
+  /** Implementor for str_to_map. */
+  private static class StringToMapImplementor extends AbstractRexCallImplementor {
+    StringToMapImplementor() {
+      super("str_to_map", NullPolicy.STRICT, false);
+    }
+
+    @Override Expression implementSafe(RexToLixTranslator translator,
+        RexCall call, List<Expression> argValueList) {
+      String defaultStringDelimiter = ",";
+      String defaultKeyValueDelimiter = ":";
+      if (argValueList.size() == 1) {
+        return Expressions.call(
+            BuiltInMethod.STR_TO_MAP.method,
+            argValueList.get(0),
+            Expressions.constant(defaultStringDelimiter),
+            Expressions.constant(defaultKeyValueDelimiter));
+      } else if (argValueList.size() == 2) {
+        return Expressions.call(
+          BuiltInMethod.STR_TO_MAP.method,
+              argValueList.get(0),
+              argValueList.get(1),
+              Expressions.constant(defaultKeyValueDelimiter));
```

Comment:
if you only use the default Strings for Expressions.constant, you could just initialize them as expressions instead?

Reply:
i am wondering whether the return statement could be extracted out, and the if-else-if (or switch, if you choose to use it) could just be used for assigning operands and whatnot. Right now it looks like a lot of duplicate code.

Reply:
done